<launch>
    <param name="/use_simulated_time" type="bool" value="true" />

    <node pkg="predictive_velocity_controller" name="panda_translator" type="panda_translator.py" output="screen" />

    <node pkg="predictive_velocity_controller" name="KCT_Controller" type="predictive_controller_node" output="screen">
        <param name="/planning_root_link" type="string" value="panda_link0" />
        <param name="/planning_tip_link" type="string" value="panda_hand" />
        <param name="/number_of_joints" type="int" value="7" />
        <param name="/joint0_name" type="string" value="panda_joint1" />
        <param name="/joint0_pos_up_limit" type="double" value="2.8973" />
        <param name="/joint0_pos_down_limit" type="double" value="2.8973" />
        <param name="/joint0_vel_limit" type="double" value="2.175" />
        <param name="/joint1_name" type="string" value="panda_joint2" />
        <param name="/joint1_pos_up_limit" type="double" value="1.7628" />
        <param name="/joint1_pos_down_limit" type="double" value="1.7628" />
        <param name="/joint1_vel_limit" type="double" value="2.175" />
        <param name="/joint2_name" type="string" value="panda_joint3" />
        <param name="/joint2_pos_up_limit" type="double" value="2.8973" />
        <param name="/joint2_pos_down_limit" type="double" value="2.8973" />
        <param name="/joint2_vel_limit" type="double" value="2.175" />
        <param name="/joint3_name" type="string" value="panda_joint4" />
        <param name="/joint3_pos_up_limit" type="double" value="0.0" />
        <param name="/joint3_pos_down_limit" type="double" value="3.0718" />
        <param name="/joint3_vel_limit" type="double" value="2.175" />
        <param name="/joint4_name" type="string" value="panda_joint5" />
        <param name="/joint4_pos_up_limit" type="double" value="2.8973" />
        <param name="/joint4_pos_down_limit" type="double" value="2.8973" />
        <param name="/joint4_vel_limit" type="double" value="2.61" />
        <param name="/joint5_name" type="string" value="panda_joint6" />
        <param name="/joint5_pos_up_limit" type="double" value="3.7525" />
        <param name="/joint5_pos_down_limit" type="double" value="0.0175" />
        <param name="/joint5_vel_limit" type="double" value="2.61" />
        <param name="/joint6_name" type="string" value="panda_joint7" />
        <param name="/joint6_pos_up_limit" type="double" value="2.8973" />
        <param name="/joint6_pos_down_limit" type="double" value="2.8973" />
        <param name="/joint6_vel_limit" type="double" value="2.61" />
        <param name="/user_model" type="int" value="0" />
        <param name="/robot_type" type="int" value="1" />
        <remap from ="/planning_robot_urdf" to="/robot_description" />
        <remap from ="/joint_commands" to="/panda_joint_commands" />
    </node>

</launch>